src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/jpa/entity/BookingEntity.kt

```diff
@@ -43,6 +43,12 @@ data class BookingEntity(
   var cancellation: CancellationEntity?,
   @OneToOne(mappedBy = "booking")
   var confirmation: ConfirmationEntity?,
+  @OneToOne
+  @JoinColumn(name = "application_id")
+  var application: ApplicationEntity?,
+  @OneToOne
+  @JoinColumn(name = "offline_application_id")
+  var offlineApplication: OfflineApplicationEntity?,
```

Comment:
are we able to ask a booking for its application in a consistent way e.g. `booking.application` which retrieves both varieties of application?

Or are there two "association" methods now? `booking.application` and `booking.offlineApplication`?

Reply:
Unfortunately not without making OfflineApplication derive from Application (so option A or B from the diagram we discussed the other day)

I think in practice though this won't make much difference as you would need to do a type check anyway, e.g.

```
if (booking.application != null) {
   // do something
} else if (booking.offlineApplication != null) {
   // do something else
} else {
   throw RuntimeException("Missing application for booking ${booking.id}")
}
```

versus

```
when (booking.application) {
  is OnlineApplicationEntity -> //do something
  is OfflineApplicationEntity -> //do something else
  null -> throw RuntimeException("Missing application for booking ${booking.id}")
}
```